.github/workflows/gradle.yml

```diff
@@ -0,0 +1,53 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: '*'
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out this repo
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        repository: ruderphilipp/jmc.git
+        ref: main
+    - name: Check out the triggering repository
+      run: |
+         declare -A SUBMODULES=( \
+            [jameica]=jameica \
+            [datasource]=lib/datasource \
+            [util]=lib/util \
+            [example]=plugins/example \
+            [hibiscus]=plugins/hibiscus \
+            )
+
+         set -x
+         REPONAME=${GITHUB_REPOSITORY#*/}
+         if [ ${REPONAME} = "jmc" ]; then
+            echo "Github Actions run started from this repository"
+         else
+            git -C ${SUBMODULES[$REPONAME]} remote add action_repo git@github.com:${GITHUB_REPOSITORY}
+            git -C ${SUBMODULES[$REPONAME]} fetch action_repo ${GITHUB_SHA}
+            git -C ${SUBMODULES[$REPONAME]} checkout ${GITHUB_SHA}
+         fi
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
```

Comment:
aktuell ist das Projekt noch auf dem Stand von Java 8 (vgl. Gradle-Konfiguration). Deswegen würde ich hier auch Version `8` eintragen, da so die Compilerversion mit dem Kompatibilitätslevel übereinstimmt.

Reply:
Stimmt. Das hatte ich aus einem deiner (?) Repositories übernommen und nicht weiter hinterfragt. Interessehalber: ich dachte, Java ist weitestgehend rückwärtskompatibel und daher könnte man mit 11 auch für 8 kompilieren. Stimmt das?

Reply:
Ja, Java ist rückwärtskompatibel. Das klappt auch alles, wenn man Version 11 oder 16 konfiguriert (habe beides ausprobiert).
Mir macht nur Sorge, dass mit Java9 ein neuer modularer Aufbau etabliert wurde. Da kann es zu kleinen gemeinen Abweichungen kommen, falls irgendetwas in ein extra Modul ausgelagert wurde. Diese Compilerfehler sind dann schwer zu debuggen.

Noch nutzt der Code kein einziges Konstrukt neuer als Java8. Ich möchte ja genau daran arbeiten, hier den Code durch die neuen Möglichkeiten zu verschlanken. Deswegen ist aktuell eine höhere Version noch nicht notwendig.

Reply:
Ist jetzt auf zu 8 geändert.